stm32-modules/heater-shaker/firmware/motor_task/motor_policy.cpp

```diff
@@ -0,0 +1,34 @@
+#include <cstdlib>
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wvolatile"
+#pragma GCC diagnostic ignored "-Wregister"
+#include "mc_interface.h"
+#include "mc_stm_types.h"
+#pragma GCC diagnostic pop
+
+#include "motor_policy.hpp"
+
+MotorPolicy::MotorPolicy(MCI_Handle_t *handle): motor_handle(handle) {}
+
+auto MotorPolicy::set_rpm(int16_t rpm) -> void {
+    int16_t current_speed = get_current_rpm();
+    int16_t command_01hz = rpm * _01HZ / _RPM;
+    uint16_t ramp_time =
+        static_cast<int32_t>(std::abs(rpm - current_speed)) * RAMP_MS_PER_RPM;
```

Comment:
I'm confused about the 3 different types here. RAMPS_MS_PER_RPM is an unsigned 32 int, the result of casting `rpm-current_speed` is a signed 32 int (makes sense) but then we're saying that the total result should be an unsigned 16 int

Reply:
that probably could be tuned up a bit. it's meant to leave space for overflows before a subsequent divide, but we've baked the divide into the rpm command, so this doesn't really need to be there. I should fix that.